docs(plugins): add DLNA Receiver plugin page (#626)
music-assistant/music-assistant.io: src/content/docs/plugins/dlna-receiver.md

---
title: "DLNA Receiver Plugin"
description: Features and Notes for the DLNA Receiver Plugin
---

# DLNA Receiver

Music Assistant can act as a UPnP/DLNA MediaRenderer on the local network. External DLNA-aware apps discover it on the network and can cast audio to any MA player.

Once an external app sends a stream URL via the standard DLNA protocol, Music Assistant proxies the audio through its streaming pipeline to the selected target player — DLNA speakers, AirPlay receivers, Chromecast, Yandex Station, Squeezebox, etc.

## Features

- Exposes Music Assistant players as DLNA MediaRenderers on the local network
- Works as either a single virtual renderer or one renderer per MA player
- Renderer names and identity persist across MA restarts, so DLNA apps remember their bookmarks
- Reports playback progress and volume back to the controlling app

> - Tested with [BubbleUPnP](https://bubblesoftapps.com/bubbleupnp/)

Reviewer: This isnt a feature

## Configuration

> The plugin is single-instance. After adding it in Music Assistant, configure the following settings:

Reviewer: ```suggestion
The plugin is single-instance. After adding it to Music Assistant, configure the following:
```

### Settings

- **Friendly name prefix** — name shown to DLNA apps on the network. When multiple players are exposed, the MA player name is appended automatically (e.g. `Music Assistant — Kitchen`). Default: `Music Assistant`.
- **Target players** — controls which Music Assistant players appear as DLNA devices:
  - *empty* — a single virtual renderer is advertised; you can route the incoming stream to any player from the MA UI after it starts.
  - *comma-separated player IDs* — one DLNA device per listed player (e.g. `kitchen,living_room`).
  - `*` — automatically expose every MA player, including ones that come online later.
- **Bind IP address** — IP address the plugin listens on. Leave empty to auto-detect.
- **HTTP base port** — TCP port used by the plugin. When multiple players are exposed, the next ports are used as well (e.g. `8298`, `8299`, `8300`, …). Default: `8298`.

## Known Issues / Notes

- To listen to the plugin audio, navigate to the desired player's NOW PLAYING view and then in the menu in the top right, select Source, and choose the desired DLNA Receiver.

> - **Docker:** `network_mode: host` is required. Bridge networking blocks DLNA discovery and control points will not see any renderer.

Reviewer: I dont think this is required to say as this is already a requirement for MA in docker generally

---
title: "DLNA Receiver Plugin"
description: Features and Notes for the DLNA Receiver Plugin
---

# DLNA Receiver

Music Assistant can act as a [UPnP/DLNA MediaRenderer](https://openconnectivity.org/developer/specifications/upnp-resources/upnp/) on the local network, so that external DLNA-aware apps can discover it over SSDP and cast audio to any MA player. Contributed and maintained by [TrudenBoy](https://github.com/TrudenBoy).

External apps send a stream URL via the standard DLNA protocol (`SetAVTransportURI` / `Play`) and Music Assistant proxies the audio through its streaming pipeline to the selected target player — DLNA speakers, AirPlay receivers, Chromecast, Yandex Station, Squeezebox, etc.

> [!CAUTION]
> This is an unofficial implementation of the UPnP/DLNA MediaRenderer server side and is not certified by the UPnP Forum / Open Connectivity Foundation.

> [!WARNING]
> SSDP relies on IPv4 multicast on `239.255.255.250:1900`. When running Music Assistant in Docker, `network_mode: host` is required — bridge networking will prevent control points from discovering the renderer.

> [!NOTE]

> > Full plugin documentation: **[trudenboy.github.io/ma-provider-dlna-receiver](https://trudenboy.github.io/ma-provider-dlna-receiver/)**

Reviewer: For this plugin which is applicable to all MA users (i.e. not just the Russian language speakers) then all relevent documentation needs to be here.

Author: Removed the link and folded the relevant bits into this page. The external site is no longer referenced.
